url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/commands/model3/Model3Controller.java

```diff
@@ -0,0 +1,725 @@
+					double jointP = stackedTrajectoryProb * bRes.probability;
+					semSentenceProbs.add(sementicString, jointP);
+					
+				}
+				
+				
+			}
+			
+			
+		}
+		
+		
+		Map<String, Double> semanticStringProbs = new HashMap<String, Double>(semSentenceProbs.size());
+		for(Map.Entry<String, Double> e : semSentenceProbs.entrySet()){
+			double normP = e.getValue() / sumTrajectoryProb;
+			semanticStringProbs.put(e.getKey(), normP);
+		}
+		
+		return semanticStringProbs;
+	}
+	
+	
+	
+	public static String getSemanticString(LiftedVarValue liftedRF, LiftedVarValue bindingConstraints){
+
+		StringBuffer buf = new StringBuffer();
+		boolean first = true;
+		for(GroundedProp gp : liftedRF.conditions){
+			if(!first){
+				buf.append(" ");
+			}
+			buf.append(gp.pf.getName());
+			for(String p : gp.pf.getParameterClasses()){
+				buf.append(" ").append(p);
+			}
+			first = false;
+		}
+		for(GroundedProp gp : bindingConstraints.conditions){
+			buf.append(" ").append(gp.pf.getName());
+			for(String p : gp.pf.getParameterClasses()){
+				buf.append(" ").append(p);
+			}
+		}
+		
+		return buf.toString();
+	}
+	
+	
+	public static List <TrainingElement> getCommandsDataset(Domain domain, String path, StateParser sp){
+		
+		TrainingElementParser teparser = new TrainingElementParser(domain, sp);
+		
+		//get dataset
+		List<TrainingElement> dataset = teparser.getTrainingElementDataset(path, ".txt");
+		
+		return dataset;
+	}
+	
+	public static List<WeightedMTInstance> getStrictMTDataset(Tokenizer tokenizer, String pathToDataDir, String dataFileExtension){
+		
+		//get rid of trailing /
+		if(pathToDataDir.charAt(pathToDataDir.length()-1) == '/'){
+			pathToDataDir = pathToDataDir.substring(0, pathToDataDir.length());
+		}
+		
+		File dir = new File(pathToDataDir);
+		final String ext = new String(dataFileExtension);
+		
+		FilenameFilter filter = new FilenameFilter() {
+			public boolean accept(File dir, String name) {
+				if(name.endsWith(ext)){
+					return true;
+				}
+				return false;
+			}
+		};
+		String[] children = dir.list(filter);
+		
+		List<WeightedMTInstance> dataset = new ArrayList<WeightedMTInstance>(children.length);
+		for(String child : children){
+			String path = pathToDataDir + "/" + child;
+			WeightedMTInstance inst = getStrictMTDataInstanceFromFile(tokenizer, path);
+			dataset.add(inst);
+		}
+		
+		
+		return dataset;
+		
+	}
+	
+	public static WeightedMTInstance getStrictMTDataInstanceFromFile(Tokenizer tokenizer, String pathToFile){
+		
+		WeightedMTInstance d = null;
+		
+		try {
+			BufferedReader in = new BufferedReader(new FileReader(pathToFile));
+			String l1 = in.readLine().trim();
+			String l2 = in.readLine().trim();
+			
+			d = new WeightedMTInstance(tokenizer.tokenize(l2));
+			d.addWeightedSemanticCommand(tokenizer.tokenize(l1), 1.);
+			
+			in.close();
+			
+		} catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)